earthfile2llb/interpreter.go

```diff
@@ -631,6 +633,22 @@ func (i *Interpreter) handleFromDockerfile(ctx context.Context, cmd spec.Command
 	return nil
 }
 
+// getWorkingDir returns the current working directory, which is the Earthfile
+// of the current target; except for COMMANDS which inherrit the caller's target
+// directory.
+func (i *Interpreter) getWorkingDir(ctx context.Context, cmd spec.Command) (string, error) {
+	n := len(i.workingDirStack)
+	if n > 0 {
+		return i.workingDirStack[n-1], nil
+	}
+
+	workingDir, err := filepath.Abs(filepath.Dir(cmd.SourceLocation.File))
+	if err != nil {
+		return "", i.wrapError(err, cmd.SourceLocation, "failed to get absolute path for %s", cmd.SourceLocation.File)
+	}
+	return workingDir, nil
+}
```

Comment:
I would pass the source file location via the build context, from the resolver. It would save us from the complexity of tracking this stack here.

It is already returned here: https://github.com/earthly/earthly/blob/main/buildcontext/resolver.go#L31. We just need to plug it into the converter.

The converter already gets the build context data in its constructor: https://github.com/earthly/earthly/blob/main/earthfile2llb/converter.go#L92.

Reply:
Would this new method also be a better way to get the working dir than what I had previously in `handleCache` below?
i.e.
```golang
dir = path.Clean(path.Join("/", i.converter.mts.Final.MainImage.Config.WorkingDir, dir))
```

Reply:
Not really... here we're interested in the dir where the Earthfile exists on the host, whereas `dir = path.Clean(path.Join("/", i.converter.mts.Final.MainImage.Config.WorkingDir, dir))` seems to be the work dir within the build environment.

Reply:
I misunderstood that, thanks

Reply:
> I would pass the source file location via the build context, from the resolver. It would save us from the complexity of tracking this stack here.

That's much simpler -- good suggestion. It turns out that the `Locally()` method took a workingDir as an argument, which is no longer needed.

> Would this new method also be a better way to get the working dir than what I had previously in handleCache below?

keeping the existing code for `handleCache` is good; `i.converter.mts.Final.MainImage.Config.WorkingDir` is relative to the containers, where as this PR only deals with the special case of local file system paths (which are required for `LOCALLY`).